### PR TITLE
Send capture method for Doc, Selfie & Liveness in sdk_metadata

### PR DESCRIPTION
**Pull request by kopijunkie**
# Problem
As an Analyst,
I want to be able to distinguish between different capture methods so that I can better understand how to improve the various product metrics

# Solution
We should capture the following properties inside `sdk_metadata`:
* isCrossDeviceFlow (true|false)
* deviceType (mobile|desktop) - _based on the isDesktop method_
* captureMethod (live|html5)

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://912-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Capture/Face.js [hunk 1]
```diff
@@ -36,13 +36,25 @@ class Face extends Component {
...
+      ...defaultPayload,
+      ...payload,
+      sdkMetadata: this.addIsCrossDeviceAndDeviceTypeProperties(payload.sdkMetadata),
+      id
+    }
+    actions.createCapture(faceCaptureData)
     nextStep()
   }
 
+  addIsCrossDeviceAndDeviceTypeProperties = sdkMetadata => ({
```
Comment (josokinas) [inline src/components/Capture/Face.js]: This seem to be repeated in both components...

Comment (kopijunkie) [reply src/components/Capture/Face.js]: Good spot! I think I'll move it out into a util under `onfidoApi.js`.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-912-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Capture/Face.js [hunk 2]
```diff
@@ -36,13 +36,25 @@ class Face extends Component {
...
+    ...sdkMetadata,
+    deviceType: isDesktop ? 'desktop' : 'mobile',
+    isCrossDeviceFlow: this.props.mobileFlow
+  })
+
   handleVideoCapture = payload => this.handleCapture({ ...payload, variant: 'video' })
 
-  handleUpload = blob => this.handleCapture({ blob })
+  handleUpload = blob => this.handleCapture({ blob, sdkMetadata: { captureMethod: 'html5' } })
```
[the remark this answers is not in the document]
Comment (kopijunkie) [reply src/components/Capture/Face.js]: I think it's worth adding, unless it's already being tracked in analytics, but not sure if we'd need to make a back end change for it also.

[the remark this answers is not in the document]
Comment (seewah) [reply src/components/Capture/Face.js]: spoken with @kopijunkie . it is not that valuable to capture the flow scenario info in the metadata for now. thanks

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-912-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://912-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**